pkg/reconciler/buildrun/resources/taskrun.go

```diff
@@ -401,6 +412,13 @@ func GenerateTaskRun(
 		}
 	}
 
+	// Setup image processing
+	buildRunOutput := buildRun.Spec.Output
+	if buildRunOutput == nil {
+		buildRunOutput = &buildv1alpha1.Image{}
+	}
+	SetupImageProcessing(expectedTaskRun, cfg, build.Spec.Output, *buildRunOutput)
```

Comment:
Can you add a comment that here is the call to the function with the main business logic to decide if a new container is added for pushing and labelling/annotations or if a new container is added just for labelling/annotations or that this might be a no-op, where nothing happens

Reply:
Done.